objects/obj_creation_popup/Draw_0.gml

```diff
@@ -110,59 +110,70 @@ if (col_shift){
+                case 5: geh = obj_creation.mobi[co, ide]; break;
+            }
+
             draw_set_halign(fa_right);
             draw_set_color(38144);
-            draw_rectangle(x5,y5,x5-string_width(string_hash_to_newline(title)),y5+string_height(string_hash_to_newline(title))-2,1);
-            draw_text(x5,y5,string_hash_to_newline(string(title)));
-            
-            if (scr_hit(x5-string_width(string_hash_to_newline(title)),y5,x5,y5+string_height(string_hash_to_newline(title))-2)=true){
-                draw_set_color(c_white);draw_set_alpha(0.2);
-                draw_rectangle(x5,y5,x5-string_width(string_hash_to_newline(title)),y5+string_height(string_hash_to_newline(title))-2,0);
-                if (obj_creation.mouse_left=1) and (obj_creation.cooldown<=0){
-                
-                    var bad;bad=0;
-                    if (type=106) and ((gg=3) or (gg=4) or (gg=5)) then bad=1;
-                    
-                    if (bad=0){
-                        obj_creation.cooldown=8000;
-                        tab=1;
-                        
-                        if (gg<4) then target_gear=gg;
-                        if (gg=4) then target_gear=5;
-                        if (gg=5) then target_gear=4;
-                        
-                        scr_weapons_equip();// Gets item list
+            draw_rectangle(x5, y5, x5 - string_width(string_hash_to_newline(title)), y5 + string_height(string_hash_to_newline(title)) - 2, 1);
+            draw_text(x5, y5, string_hash_to_newline(string(title)));
+
+            if (scr_hit(x5 - string_width(string_hash_to_newline(title)), y5, x5, y5 + string_height(string_hash_to_newline(title)) - 2)) {
+                draw_set_color(c_white);
+                draw_set_alpha(0.2);
+                draw_rectangle(x5, y5, x5 - string_width(string_hash_to_newline(title)), y5 + string_height(string_hash_to_newline(title)) - 2, 0);
+
+                if (obj_creation.mouse_left == 1 && obj_creation.cooldown <= 0) {                   
+                    var unit_type = type - 100;
+                    var is_invalid = (
+                        unit_type == eROLE.Dreadnought &&
+                        (
+                            // slots
+                            gg == 3 ||
+                            gg == 4 ||
+                            gg == 5
+                        )
+                    );
+
+                    if (!is_invalid) {
+                        obj_creation.cooldown = 8000;
+
+                        item_name = [];
+                        scr_get_item_names(
+                            item_name,
+                            unit_type, // eROLE
+                            gg, // slot
+                            eENGAGEMENT.Ranged,
+                            false,  // no company standard
+                            false   // don't limit to available items
+                        );
                     }
```

Comment:
I think this should solve the creation screen role equip issue.
```suggestion
                    if (!is_invalid) {
                        obj_creation.cooldown = 8000;
                        tab=1;
                        target_gear=gg;
                        item_name = [];
                        scr_get_item_names(
                            item_name,
                            unit_type, // eROLE
                            gg, // slot
                            eENGAGEMENT.Ranged,
                            false,  // no company standard
                            false   // don't limit to available items
                        );
                    }
```

Reply:
this fixes it. 👍